s3torchconnector/tst/unit/test_s3iterable_dataset.py

```diff
@@ -218,6 +218,387 @@ def test_from_prefix_seek_no_head():
     head_object.assert_not_called()
 
 
+"""
```

Comment:
Would some property based tests be useful here? 

Expected_keys is effectively calculated as
```
all_keys[(worker_id*num_workers)+rank::world_size*num_workers]
```

Reply:
Replaced with property based, but have to change what is tested. Now it test that all keys are evenly distributed between ranks without duplication.